### PR TITLE
Update hopelessly outdated build documentation on master

### LINKED ISSUE #384: User manual - Building on Windows - obsoleted way
https://github.com/OpenSCAP/openscap/blob/maint-1.0/docs/manual/manual.adoc#54-building-openscap-on-windows

1) we should not refer to `git://git.fedorahosted.org/git/openscap.git`, but rather to github
2) instead of `-b portable` should be probably `-b master`

We should ensure, that we can (still) can build it on Windows

### LINKED ISSUE #1011: Remove or move Cygwin Instructions from the user manual
We enabled to build OpenSCAP on Windows using Microsoft Visual Studio 2017. I think this will be the preferred way for our contributors who will contribute on Windows.

Do we want to keep the instructions that say it is best to build it without probes using Cygwin?

https://github.com/OpenSCAP/openscap/blob/master/docs/manual/manual.adoc#building-openscap-on-windows

I suggest removing these or at least moving them down so that Building Using Visual Studio section is visible at first place. 

https://github.com/OpenSCAP/openscap/blob/master/docs/manual/manual.adoc#67-building-openscap-on-windows-using-visual-studio

What is your opinion on this topic?

### LINKED ISSUE #1066: Update 'Testing library' section in manual
OpenSCAP in master doesn't build probes as separate binaries anymore.
`OVAL_PROBE_DIR` which specified where to source the binaries from is removed in #1050.

Remove mention of `OVAL_PROBE_DIR` in manual and update how to test the library.

### LINKED ISSUE #1069: Build documentation hopelessly outdated on master
#### Description of Problem:
Master is built using cmake but there are tons of mentions of autogen, automake, etc... still in master. All the docs need to be updated.

Furthermore, Windows is considered second citizen platform (there is a special chapter for it and it says "openscap can also be built on windows...") which is not the case any-more. The README assumes you are using Linux and has no mention of Windows or OSX. We need to restructure the docs to support Linux, MacOS X and Windows builds equally.

I am mainly talking about `docs/manual/manual.adoc` and README here but please... `grep` the whole repo when fixing this.

#### OpenSCAP Version:
```
git show
commit 0f648dc2d6efe90b9519a333dd25de697c008f1d (HEAD -> python2_disabled_if_not_found, upstream/master, master)
Merge: 5fdc2ad78 9239c06b5
Author: Watson Yuuma Sato <wsato@redhat.com>
Date:   Fri Jun 15 13:26:29 2018 +0200

    Merge pull request #1050 from cipherboy/rm-oval-probe-dir

    Remove occurrences of OVAL_PROBE_DIR in master
```

#### Operating System & Version:
Vokna 10

### PR DESCRIPTION
**Pull request by jan-cerny**
In this PR I am going to provide patches that improve our build documentation. These improvements will reflect the recent changes in master branch, namely CMake and Windows.

I have:
- split User Manual to User Manual and Developer Manual
- unified the build documentation from the README and from the User Manual
- moved the build instructions from README to Developer Manual
- Updated the build instructions on Linux
- Removed non-applicable parts of the Developer Manual
- Removed Cygwin instructions
- Improved Asciidoc syntax
- Improved the logic of text flow
- fixed typos

Fixes #1069.
Fixes #1066 
Fixes #1011
Fixes #384

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,89 +10,9 @@ The oscap program is a command line tool that allows users to load, scan,
...
-Not every check tests the oscap tool, however, when the CUSTOM_OSCAP variable is set, only the checks which do are executed.
-
-
-4) Run the installation procedure by executing the following command:
-```
-make install
-```
-
+ * Manual: [Oscap User Manual](https://github.com/OpenSCAP/openscap/blob/master/docs/manual/manual.adoc)
+ * Compilation, testing and debugging: [OpenSCAP Developer Manual](https://github.com/OpenSCAP/openscap/blob/master/docs/developer/developer.adoc)
```
Comment (mpreisler) [inline README.md]: this link is absolute, link to the relative path. that way the link will work even if we branch

File: README.md [hunk 2]
```diff
@@ -10,89 +10,9 @@ The oscap program is a command line tool that allows users to load, scan,
...
+ * For new contributors: [How to contribute](https://github.com/OpenSCAP/openscap/blob/master/docs/contribute/contribute.adoc)
```
Comment (mpreisler) [inline README.md]: same here

**Comment by jan-cerny** [comment]
Hi, what is your opinion on changes proposed in this PR? Is this outline a way to go? Do you suggest a different structure? Should the build instructions be more detailed? Have you missed any important information in your previous experience?

File: docs/developer/developer.adoc [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+ $ git clone https://github.com/OpenSCAP/openscap.git
+ $ cd openscap
++
+
+. *Get the build dependencies*
++
+--
+To build the library you will also need the following build dependencies.
+Build dependencies may vary depending on enabled features (by the ```cmake``` command).
+By default, OpenSCAP requires these dependencies to build:
```
Comment (matejak) [inline docs/developer/developer.adoc]: I think that the list of dependencies is not needed, it spans huge amount of vertical space, individual items have platform-dependent names, and we repeat everything below, where we tell people how to install those.

File: docs/developer/developer.adoc [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+ $ cmake ../
+ $ make
+
+
+. *Run the tests*
++
+After building the library you might want to run library self-checks. To do
+that you need to have these additional packages installed:
+
+ wget lua which procps-ng initscripts chkconfig sendmail
```
Comment (matejak) [inline docs/developer/developer.adoc]: Are we sure about `chkconfig` and `initscripts`?

File: docs/developer/developer.adoc [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
++
+and it is also required to have `sendmail` service running on the system:
+
+ $ systemctl start sendmail.service
++
+Now you can execute the following command to run library self-checks:
+
+ $ ctest
++
+Note: If you want to run `make distcheck` you will also need to install
```
Comment (matejak) [inline docs/developer/developer.adoc]: I don't think that `make distcheck` is relevant in the `master` branch.

**Comment by matejak** [comment]
This is a great PR, which shows our users and potential contributors the right information they need.
I have posted one suggestion and two questions.

File: docs/developer/developer.adoc (hunk 3 above)
Comment (jan-cerny) [reply docs/developer/developer.adoc]: +1

File: docs/developer/developer.adoc (hunk 4 above)
Comment (jan-cerny) [reply docs/developer/developer.adoc]: I hope yes, this was introduced by https://github.com/OpenSCAP/openscap/pull/788

File: docs/developer/developer.adoc (hunk 5 above)
Comment (jan-cerny) [reply docs/developer/developer.adoc]: Good point. In fact, this paragraph speaks about installing `asciidoctor`. `asciidoctor` is a tools that converts `.adoc` files to HTML, including this document. Generating HTML docs used to be a part of `make dist` in `maint-1.2`. We forgot to port it over to `master` when we have ported OpenSCAP to CMake. See https://github.com/OpenSCAP/openscap/issues/1083
Actually, I think we need to mention `asciidoctor` at other, place, most likely in the "Install" section.

File: docs/developer/developer.adoc (hunk 4 above)
Comment (matejak) [reply docs/developer/developer.adoc]: I see, thanks for pointing that out.

**Comment by matejak** [comment]
Thanks again for this PR!

**Pull request merged**